includes/Core/Util/User_Input_Settings.php

```diff
@@ -128,18 +128,22 @@ function( $setting ) {
 	private function sync_with_proxy( $settings = null ) {
 		$user_input_settings_url = $this->authentication->get_google_proxy()->url( Google_Proxy::USER_INPUT_SETTINGS_URI );
 		$creds                   = $this->authentication->credentials()->get();
+		$access_token            = $this->authentication->get_oauth_client()->get_access_token();
 
 		$user_input_settings_args = array(
 			'headers' => array(
-				'Authorization' => 'Bearer ' . $this->authentication->get_oauth_client()->get_access_token(),
-				'Content-Type'  => 'application/json',
+				'Content-Type' => 'application/json',
 			),
 			'body'    => array(
 				'site_id'     => $creds['oauth2_client_id'],
 				'site_secret' => $creds['oauth2_client_secret'],
 			),
 		);
 
+		if ( ! empty( $access_token ) && is_string( $access_token ) ) {
+			$user_input_settings_args['headers']['Authorization'] = 'Bearer ' . $access_token;
+		}
+
```

Comment:
Why change this? The access token is required for making this request, is it not?

Reply:
Had to change it because we don't use real OAuth client during e2e tests, thus it doesn't return an access token and causes unexpected issues.

> The access token is required for making this request, is it not?

Yes, it is required by the proxy server itself. The user input settings class just passes what it has.

Reply:
Hmm, okay. I suppose this is fine then since we weren't guarding against making the request if there was no token before.